### ISSUE TITLE
Pretraining Time

### ISSUE THREAD

**Issue by wangjk666** [salesforce/BLIP#19]
Hi, it is mentioned in the paper that you pretrain the model on two 16-GPU nodes with batch-size=2880, I wonder how long does it take to pretrain on the filtered CC + COCO + SBU + VG dataset, and could you share your log with me? 

Thanks a lot.

**Comment by LiJunnan1992** [comment]
Hi, the pre-training takes 3-4 days using the CapFilt version of CC+COCO+SBU+VG, here is the log:
```
{"train_lr": "0.000", "train_loss_ita": "3.002", "train_loss_itm": "0.337", "train_loss_lm": "3.352", "epoch": 0}
{"train_lr": "0.000", "train_loss_ita": "2.425", "train_loss_itm": "0.256", "train_loss_lm": "2.841", "epoch": 1}
{"train_lr": "0.000", "train_loss_ita": "2.542", "train_loss_itm": "0.215", "train_loss_lm": "2.724", "epoch": 2}
{"train_lr": "0.000", "train_loss_ita": "2.414", "train_loss_itm": "0.196", "train_loss_lm": "2.651", "epoch": 3}
{"train_lr": "0.000", "train_loss_ita": "2.400", "train_loss_itm": "0.170", "train_loss_lm": "2.598", "epoch": 4}
{"train_lr": "0.000", "train_loss_ita": "2.451", "train_loss_itm": "0.148", "train_loss_lm": "2.556", "epoch": 5}
{"train_lr": "0.000", "train_loss_ita": "2.324", "train_loss_itm": "0.171", "train_loss_lm": "2.521", "epoch": 6}
{"train_lr": "0.000", "train_loss_ita": "2.138", "train_loss_itm": "0.170", "train_loss_lm": "2.488", "epoch": 7}
{"train_lr": "0.000", "train_loss_ita": "2.284", "train_loss_itm": "0.137", "train_loss_lm": "2.459", "epoch": 8}
{"train_lr": "0.000", "train_loss_ita": "2.172", "train_loss_itm": "0.166", "train_loss_lm": "2.434", "epoch": 9}
{"train_lr": "0.000", "train_loss_ita": "2.050", "train_loss_itm": "0.160", "train_loss_lm": "2.412", "epoch": 10}
{"train_lr": "0.000", "train_loss_ita": "1.949", "train_loss_itm": "0.164", "train_loss_lm": "2.392", "epoch": 11}
{"train_lr": "0.000", "train_loss_ita": "1.730", "train_loss_itm": "0.171", "train_loss_lm": "2.373", "epoch": 12}
{"train_lr": "0.000", "train_loss_ita": "1.631", "train_loss_itm": "0.169", "train_loss_lm": "2.357", "epoch": 13}
{"train_lr": "0.000", "train_loss_ita": "1.583", "train_loss_itm": "0.169", "train_loss_lm": "2.344", "epoch": 14}
{"train_lr": "0.000", "train_loss_ita": "1.534", "train_loss_itm": "0.171", "train_loss_lm": "2.329", "epoch": 15}
{"train_lr": "0.000", "train_loss_ita": "1.496", "train_loss_itm": "0.169", "train_loss_lm": "2.317", "epoch": 16}
{"train_lr": "0.000", "train_loss_ita": "1.468", "train_loss_itm": "0.166", "train_loss_lm": "2.305", "epoch": 17}
{"train_lr": "0.000", "train_loss_ita": "1.440", "train_loss_itm": "0.163", "train_loss_lm": "2.295", "epoch": 18}
{"train_lr": "0.000", "train_loss_ita": "1.421", "train_loss_itm": "0.160", "train_loss_lm": "2.286", "epoch": 19}
```

**Comment by wangjk666** [comment]
OK. Thanks a lot.

**Comment by wangjk666** [comment]
Hi, I pre-train the model on the CapFilt version (captioner-L + filter) of CC+COCO+SBU+VG, the batch size is 2880, and the number of training samples is 25154458 (~25M), however, the estimated time is 13 hours x 20 epochs, which is far longer than what you described. I wonder if there might be some misunderstandings or some tricks to save pre-training time?

Thanks.

**Comment by LiJunnan1992** [comment]
What GPU do you use?

**Comment by wangjk666** [comment]
Tesla A100 4 nodes

**Comment by LiJunnan1992** [comment]
If you use multi-node training, inter-node communication could be a speed bottleneck.

btw, I haven't used any other tricks.

**Comment by wangjk666** [comment]
Ok, thanks a lot.

**Comment by dyashuni** [comment]
> Hi, the pre-training takes 3-4 days using the CapFilt version of CC+COCO+SBU+VG, here is the log:
> 
> ```
> {"train_lr": "0.000", "train_loss_ita": "3.002", "train_loss_itm": "0.337", "train_loss_lm": "3.352", "epoch": 0}
> {"train_lr": "0.000", "train_loss_ita": "2.425", "train_loss_itm": "0.256", "train_loss_lm": "2.841", "epoch": 1}
> {"train_lr": "0.000", "train_loss_ita": "2.542", "train_loss_itm": "0.215", "train_loss_lm": "2.724", "epoch": 2}
> {"train_lr": "0.000", "train_loss_ita": "2.414", "train_loss_itm": "0.196", "train_loss_lm": "2.651", "epoch": 3}
> {"train_lr": "0.000", "train_loss_ita": "2.400", "train_loss_itm": "0.170", "train_loss_lm": "2.598", "epoch": 4}
> {"train_lr": "0.000", "train_loss_ita": "2.451", "train_loss_itm": "0.148", "train_loss_lm": "2.556", "epoch": 5}
> {"train_lr": "0.000", "train_loss_ita": "2.324", "train_loss_itm": "0.171", "train_loss_lm": "2.521", "epoch": 6}
> {"train_lr": "0.000", "train_loss_ita": "2.138", "train_loss_itm": "0.170", "train_loss_lm": "2.488", "epoch": 7}
> {"train_lr": "0.000", "train_loss_ita": "2.284", "train_loss_itm": "0.137", "train_loss_lm": "2.459", "epoch": 8}
> {"train_lr": "0.000", "train_loss_ita": "2.172", "train_loss_itm": "0.166", "train_loss_lm": "2.434", "epoch": 9}
> {"train_lr": "0.000", "train_loss_ita": "2.050", "train_loss_itm": "0.160", "train_loss_lm": "2.412", "epoch": 10}
> {"train_lr": "0.000", "train_loss_ita": "1.949", "train_loss_itm": "0.164", "train_loss_lm": "2.392", "epoch": 11}
> {"train_lr": "0.000", "train_loss_ita": "1.730", "train_loss_itm": "0.171", "train_loss_lm": "2.373", "epoch": 12}
> {"train_lr": "0.000", "train_loss_ita": "1.631", "train_loss_itm": "0.169", "train_loss_lm": "2.357", "epoch": 13}
> {"train_lr": "0.000", "train_loss_ita": "1.583", "train_loss_itm": "0.169", "train_loss_lm": "2.344", "epoch": 14}
> {"train_lr": "0.000", "train_loss_ita": "1.534", "train_loss_itm": "0.171", "train_loss_lm": "2.329", "epoch": 15}
> {"train_lr": "0.000", "train_loss_ita": "1.496", "train_loss_itm": "0.169", "train_loss_lm": "2.317", "epoch": 16}
> {"train_lr": "0.000", "train_loss_ita": "1.468", "train_loss_itm": "0.166", "train_loss_lm": "2.305", "epoch": 17}
> {"train_lr": "0.000", "train_loss_ita": "1.440", "train_loss_itm": "0.163", "train_loss_lm": "2.295", "epoch": 18}
> {"train_lr": "0.000", "train_loss_ita": "1.421", "train_loss_itm": "0.160", "train_loss_lm": "2.286", "epoch": 19}
> ```

@LiJunnan1992 how did you calculate the losses in this log? Is it the average loss for each epoch?

**Comment by dyashuni** [comment]
@LiJunnan1992 I found the answer, never mind

**Comment by SCZwangxiao** [comment]
> Tesla A100 4 nodes

Did you apply some fast inter-node communication hardwares like Infiniband?

**Comment by aries-young** [comment]
> the CapFilt version of CC+COCO+SBU+VG

hello, I wonder which batch_size you choose for your pre-training? In configs/pretrain.yaml batch_size = 75, but in paper batch_size = 2880 for vit-b

**Comment by aries-young** [comment]
> Hi, the pre-training takes 3-4 days using the CapFilt version of CC+COCO+SBU+VG, here is the log:
> 
> ```
> {"train_lr": "0.000", "train_loss_ita": "3.002", "train_loss_itm": "0.337", "train_loss_lm": "3.352", "epoch": 0}
> {"train_lr": "0.000", "train_loss_ita": "2.425", "train_loss_itm": "0.256", "train_loss_lm": "2.841", "epoch": 1}
> {"train_lr": "0.000", "train_loss_ita": "2.542", "train_loss_itm": "0.215", "train_loss_lm": "2.724", "epoch": 2}
> {"train_lr": "0.000", "train_loss_ita": "2.414", "train_loss_itm": "0.196", "train_loss_lm": "2.651", "epoch": 3}
> {"train_lr": "0.000", "train_loss_ita": "2.400", "train_loss_itm": "0.170", "train_loss_lm": "2.598", "epoch": 4}
> {"train_lr": "0.000", "train_loss_ita": "2.451", "train_loss_itm": "0.148", "train_loss_lm": "2.556", "epoch": 5}
> {"train_lr": "0.000", "train_loss_ita": "2.324", "train_loss_itm": "0.171", "train_loss_lm": "2.521", "epoch": 6}
> {"train_lr": "0.000", "train_loss_ita": "2.138", "train_loss_itm": "0.170", "train_loss_lm": "2.488", "epoch": 7}
> {"train_lr": "0.000", "train_loss_ita": "2.284", "train_loss_itm": "0.137", "train_loss_lm": "2.459", "epoch": 8}
> {"train_lr": "0.000", "train_loss_ita": "2.172", "train_loss_itm": "0.166", "train_loss_lm": "2.434", "epoch": 9}
> {"train_lr": "0.000", "train_loss_ita": "2.050", "train_loss_itm": "0.160", "train_loss_lm": "2.412", "epoch": 10}
> {"train_lr": "0.000", "train_loss_ita": "1.949", "train_loss_itm": "0.164", "train_loss_lm": "2.392", "epoch": 11}
> {"train_lr": "0.000", "train_loss_ita": "1.730", "train_loss_itm": "0.171", "train_loss_lm": "2.373", "epoch": 12}
> {"train_lr": "0.000", "train_loss_ita": "1.631", "train_loss_itm": "0.169", "train_loss_lm": "2.357", "epoch": 13}
> {"train_lr": "0.000", "train_loss_ita": "1.583", "train_loss_itm": "0.169", "train_loss_lm": "2.344", "epoch": 14}
> {"train_lr": "0.000", "train_loss_ita": "1.534", "train_loss_itm": "0.171", "train_loss_lm": "2.329", "epoch": 15}
> {"train_lr": "0.000", "train_loss_ita": "1.496", "train_loss_itm": "0.169", "train_loss_lm": "2.317", "epoch": 16}
> {"train_lr": "0.000", "train_loss_ita": "1.468", "train_loss_itm": "0.166", "train_loss_lm": "2.305", "epoch": 17}
> {"train_lr": "0.000", "train_loss_ita": "1.440", "train_loss_itm": "0.163", "train_loss_lm": "2.295", "epoch": 18}
> {"train_lr": "0.000", "train_loss_ita": "1.421", "train_loss_itm": "0.160", "train_loss_lm": "2.286", "epoch": 19}
> ```

hello, I wonder which batch_size you choose for your pre-training? In configs/pretrain.yaml batch_size = 75, but in paper batch_size = 2880 for vit-b
